### PR TITLE
Added a new plot option to create a luminosity function

### PR DESCRIPTION
**Pull request by MatthieuSchaller**

### CODE REVIEW

**Comment by MatthieuSchaller** [comment]
Actually, I am not happy with the y-axis units. I want an additional `mag^-1` as part of the unit. I need to figure out how to cram it in.

File: velociraptor/autoplotter/plot.py [hunk 1]
```diff
@@ -87,6 +87,32 @@ def mass_function(
...
+
+    centers, luminosity_function, error, *_ = luminosity_function.output
+
+    # A Bit of a dirty hack but we want add a formally-dimensionless magnitude unit to the y-axis
+    unyt.unit_registry.default_unit_registry.add(
+        "mag", 1.0, dimensions=unyt.dimensions.dimensionless
+    )
+    luminosity_function = unyt.unyt_array(
+        luminosity_function.v,
+        str(luminosity_function.units) + " * mag**-1",
```
Comment (JBorrow) [inline velociraptor/autoplotter/plot.py:107]:
```suggestion
        luminosity_function.units / unyt.mag,
```

**Review by JBorrow — COMMENTED** [review]
Looks good apart from one conversion via strings.

File: velociraptor/autoplotter/plot.py (hunk 1 above)
Comment (MatthieuSchaller) [reply velociraptor/autoplotter/plot.py:107]: Will that work if mag is not a default unyt unit?

Comment (JBorrow) [reply velociraptor/autoplotter/plot.py:107]: Ah I see what you've done here, sorry. I didn't read it correctly.

You should add your unit in `https://github.com/SWIFTSIM/velociraptor-python/blob/master/velociraptor/__init__.py`. 
```
In [1]: import unyt

In [4]: unyt.define_unit("example_unit", 1.0 * unyt.kg, tex_repr=r"E")

In [5]: unyt.example_unit
Out[5]: example_unit
```

Whilst you're there, can you add a 1.0 * (base unit) to the ones for msun and hydrogen mass?

Comment (MatthieuSchaller) [reply velociraptor/autoplotter/plot.py]: That's now done. And works correctly. Not sure why my original version did not like just using `define_unit()`

**Pull request merged**